modules/product/src/main/java/com/opengamma/strata/product/SecurityIdTrade.java

```diff
@@ -0,0 +1,551 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableDefaults;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.market.ReferenceData;
+
+/**
+ * A trade representing the purchase or sale of a security,
+ * where the security is referenced by identifier.
+ * <p>
+ * This trade represents a trade in a security, defined by a quantity and price.
+ * The security is referenced using {@link SecurityId}.
+ * The identifier may be looked up in {@link ReferenceData}.
+ * <p>
+ * The reference may be to any kind of security, including equities,
+ * bonds and exchange traded derivatives (ETD).
+ */
+@BeanDefinition(constructorScope = "package")
+public final class SecurityIdTrade
```

Comment:
As I mentioned previously, more than one concept wants the name `SecurityTrade`. I stuck with what I felt was the most obvious and similar to today - `SecurityTrade` for the interface and `SecurityIdTrade` for the identifier-based class. It has the advantage that it is obviously identifier-based.